components/kyma-environment-broker/internal/vmauthoverrides/vmauthoverrides.go

```diff
@@ -0,0 +1,190 @@
+
+func (c *Client) InstallRelease(chartDir, releaseName string, namespace string, overrides map[string]interface{}, profile string) (*release.Release, error) {
+	cfg, err := c.newActionConfig(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	installAction := action.NewInstall(cfg)
+	installAction.ReleaseName = releaseName
+	installAction.Namespace = namespace
+	installAction.Timeout = 6 * time.Minute
+	installAction.CreateNamespace = true
+	installAction.Wait = true
+
+	chart, err := loader.Load(chartDir)
+	if err != nil {
+		return nil, err
+	}
+
+	profileValues, err := getProfileValues(*chart, profile)
+	if err != nil {
+		return nil, err
+	}
+	if profileValues != nil {
+		MergeMaps(overrides, profileValues)
+	}
+
+	reponse, err := installAction.Run(chart, overrides)
+	if err != nil {
+		return nil, err
+	}
+
+	return reponse, nil
+}
+
+func (c *Client) UpgradeRelease(chartDir, releaseName string, namespace string, overrides map[string]interface{}, profile string) (*release.Release, error) {
+	cfg, err := c.newActionConfig(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeAction := action.NewUpgrade(cfg)
+	upgradeAction.Timeout = 6 * time.Minute
+
+	chart, err := loader.Load(chartDir)
+	if err != nil {
+		return nil, err
+	}
+
+	profileValues, err := getProfileValues(*chart, profile)
+	if err != nil {
+		return nil, err
+	}
+	if profileValues != nil {
+		MergeMaps(overrides, profileValues)
+	}
+
+	response, err := upgradeAction.Run(releaseName, chart, overrides)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, err
+}
+
+func (c *Client) UninstallRelease(releaseName string, namespace string) (*release.UninstallReleaseResponse, error) {
+	cfg, err := c.newActionConfig(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	uninstallAction := action.NewUninstall(cfg)
+	uninstallAction.Timeout = 6 * time.Minute
+	response, err := uninstallAction.Run(releaseName)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, err
+}
+
+func (c *Client) newActionConfig(namespace string) (*action.Configuration, error) {
+	config := c.k8sconfig
+	kubeConfig := genericclioptions.NewConfigFlags(false)
+	kubeConfig.APIServer = &config.Host
+	kubeConfig.BearerToken = &config.BearerToken
+	kubeConfig.CAFile = &config.CAFile
+
+	cfg := new(action.Configuration)
+	if err := cfg.Init(kubeConfig, namespace, Driver, klog.Infof); err != nil {
+		return cfg, err
+	}
+
+	return cfg, nil
+}
+
+func GeneratePassword(isPresent bool, operation internal.ProvisioningOperation) string {
+	var VMPassword = ""
+	if isPresent {
+		VMPassword = operation.VMUser.Password
+	} else {
+		s1 := strings.Split(operation.ProvisioningParameters.ErsContext.GlobalAccountID, "-")
+		s2 := strings.Split(operation.InstanceDetails.SubAccountID, "-")
+		s3 := strings.Split(operation.InstanceID, "-")
+		VMPassword = s1[0] + s2[0] + s3[0]
+	}
+	return VMPassword
+}
+
+func getProfileValues(chart chart.Chart, profileName string) (map[string]interface{}, error) {
```

Comment:
This chart will never have profile preset values, it's only true for kyma component charts. We can remove this code as well as the MergeMaps utility.